### PR TITLE
UPD: minor changes of binary classification page

### PR DESCRIPTION
**Pull request by thibaultcordier**
# Description

Add minor changes of binary classification page.

## Type of change

- This change requires a documentation update

# Checklist

- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `make lint`
- [x] Typing passes successfully : `make type-check`
- [x] Unit tests pass successfully : `make tests`
- [x] Coverage is 100% : `make coverage`
- [x] Documentation builds successfully : `make doc`

### CODE REVIEW

File: doc/theoretical_description_binary_classification.rst [hunk 1]
```diff
@@ -6,41 +6,21 @@
...
 
-    PSs and CIs are only ‘informative’ if the sets or intervals produced by them are small. To quantify
-    this, we measure CIs using their width (denoted as :math:`|C(.)|)`, and PSs using their diameter (defined as
-    the width of the convex hull of the PS). For example, in the case of binary classification, the diameter
-    of a PS is :math:`1` if the prediction set is :math:`\{0,1\}`, and :math:`0` otherwise (since :math:`Y\in\{0,1\}`
-    always holds, the set :math:`\{0,1\}` is ‘uninformative’). A short CI such as :math:`[0.39, 0.41]`
-    is more informative than a wider one such as :math:`[0.3, 0.5]`.
+Prediction sets can be computed in the same way for multiclass and binary classification with
+:class:`~mapie.calibration.MapieClassifier`, and there are the same theoretical guarantees.
+Nevertheless, prediction sets are often much less informative in the binary case than in the multiclass case[#]_.
```
Comment (LacombeLouis) [inline doc/theoretical_description_binary_classification.rst:16]: What is [#]_?

File: doc/theoretical_description_binary_classification.rst [hunk 2]
```diff
@@ -6,41 +6,21 @@
...
-  can be interpreted as estimators of the confidence level.
-* *Set Prediction* is based on the estimation not of a single prediction,
-  but of a range of more or less precise predictions, each with a degree of confidence.
-* In contrast, *Probabilistic Prediction* is based on the estimation not of a single probability
-  of occurrence of a target or class, but of a range of more or less precise probabilities of occurrence,
-  each with a degree of confidence.
+* *Calibration* is useful for transforming a score (typically given by an ML model)
+  into the probability of making a good prediction.
+* *Set Prediction* gives the set of likely predictions with a probabilisic guarantee that the true label is in this set.
+* In contrast, *Probabilistic Prediction* gives a confidence interval for the predictive distribution.
```
Comment (LacombeLouis) [inline doc/theoretical_description_binary_classification.rst:23]: I would not start with "in contrast"

**Review by LacombeLouis — COMMENTED** [review]
Some small typo changes are required. Otherwise, this is good.

File: doc/theoretical_description_binary_classification.rst (hunk 1 above)
Comment (thibaultcordier) [reply doc/theoretical_description_binary_classification.rst:16]: It is for footnote in the documentation.

Comment (LacombeLouis) [reply doc/theoretical_description_binary_classification.rst:16]: I think this isn't clear. Maybe another way to show this?

File: doc/theoretical_description_binary_classification.rst [hunk 3]
```diff
@@ -6,41 +6,21 @@
 Theoretical Description
 =======================
```
Comment (LacombeLouis) [inline doc/theoretical_description_binary_classification.rst:8]: Global remark : in this part, to talk about the model we use $f$ while everywhere else in the documentation we talk about $\hat{\mu}$. We should uniformize

File: doc/theoretical_description_binary_classification.rst (hunk 1 above)
Comment (thibaultcordier) [reply doc/theoretical_description_binary_classification.rst]: I will remove it next to its call.

File: doc/theoretical_description_binary_classification.rst [hunk 4]
```diff
@@ -77,6 +66,9 @@ In the framework of conformal prediction, the Venn predictor has this property.
 3. Calibration
 --------------
 
+Usually, calibration is understood as perfect calibration meaning (see :doc:`theoretical_description_calibration`).
+In practice, it is more reasonable to consider approximate calibration.
+
 Definition 3 (Approximate calibration) [1].
     Fix a predictor :math:`f:\mathcal{X} \to [0, 1]` and let :math:`(\mathcal{X}, \mathcal{Y}) \sim P`.
```
Comment (LacombeLouis) [inline doc/theoretical_description_binary_classification.rst:73]:
```suggestion
    Fix a predictor :math:`\hat{\mu}:\mathcal{X} \to [0, 1]` and let :math:`(\mathcal{X}, \mathcal{Y}) \sim P`.
```

File: doc/theoretical_description_binary_classification.rst (hunk 3 above)
Comment (thibaultcordier) [reply doc/theoretical_description_binary_classification.rst:8]: Done.

File: doc/theoretical_description_binary_classification.rst [hunk 5]
```diff
@@ -77,6 +66,9 @@ In the framework of conformal prediction, the Venn predictor has this property.
...
     The predictor :math:`f:\mathcal{X} \to [0, 1]` is :math:`(\epsilon,\alpha)`-calibrated
```
Comment (LacombeLouis) [inline doc/theoretical_description_binary_classification.rst:74]:
```suggestion
    The predictor :math:`\hat{\mu}:\mathcal{X} \to [0, 1]` is :math:`(\epsilon,\alpha)`-calibrated
```

File: doc/theoretical_description_binary_classification.rst [hunk 6]
```diff
@@ -6,23 +6,16 @@
...
+There are mainly three different ways to handle uncertainty quantification in binary classification:
+calibration (see :doc:`theoretical_description_calibration`), confidence interval (CI) for the probability
+:math:`P(Y \vert \hat{\mu}(X))` and prediction sets (see :doc:`theoretical_description_classification`).
+These 3 notions are tightly related for score-based classifier, as it is shown in [1]. 
+
+Prediction sets can be computed in the same way for multiclass and binary classification with
+:class:`~mapie.calibration.MapieClassifier`, and there are the same theoretical guarantees.
+Nevertheless, prediction sets are often much less informative in the binary case than in the multiclass case [#]_.
+
+.. [#] From Gupta et al [1]:
```
Comment (LacombeLouis) [inline doc/theoretical_description_binary_classification.rst:18]: I think there is a problem here. Either the footnote is at the bottom of the page (it was the case before, but its number was not displayed, that's why we asked it to be changed), either you include it directly in the main text

Comment (thibaultcordier) [reply doc/theoretical_description_binary_classification.rst:18]: It was intentional in fact.

Comment (thibaultcordier) [reply doc/theoretical_description_binary_classification.rst:18]: I will find another way to quote this part.

**Review by vincentblot28 — APPROVED** [review]

**Pull request merged**